js/common/modules/@arangodb/arango-collection-common.js

```diff
@@ -322,8 +321,8 @@ ArangoCollection.prototype.iterate = function (iterator, options) {
     if (probability >= 1.0) {
       cursor = this.all();
     }else {
-      stmt = sprintf('FOR d IN %s FILTER rand() >= @prob RETURN d', this.name());
-      stmt = db._createStatement({ query: stmt });
+      stmt = sprintf('FOR d IN %s FILTER rand() <= @prob RETURN d', this.name());
+      stmt = arangodb.db._createStatement({ query: stmt });
```

Comment:
But why does it different?

It looks absolutely same

Reply:
db is defined, yet its value is `undefined`. I also tried `print(require("@arangodb").db))` and it logged `undefined`. I'm puzzled why that is.

Reply:
I mean you remove line
var db = arangodb.db; and write it inplace.